pygmt/tests/test_text.py

```diff
@@ -92,10 +93,48 @@ def test_text_nonexistent_filename():
     Run text by passing in a list of filenames with one that does not exist
     """
     fig = Figure()
-    with pytest.raises(GMTInvalidInput):
+    with pytest.raises(GMTCLibError):
         fig.text(region=[10, 70, -5, 10], textfiles=[POINTS_DATA, "notexist.txt"])
 
 
+@pytest.mark.mpl_image_compare
+def test_text_position(region):
+    """
+    Print text at center middle and four corners (top left/right, bottom
+    left/right) of map.
+    """
+    fig = Figure()
+    fig.text(region=region, projection="x1c", frame="a", position="CM", text="C M")
+    for position in ("TL", "TR", "BL", "BR"):
+        fig.text(frame="a", position=position, text=position)
+    return fig
+
+
+def test_text_xy_with_position_fails(region):
+    """
+    Run text by providing both x/y pairs and position arguments.
+    """
+    fig = Figure()
+    with pytest.raises(GMTInvalidInput):
+        fig.text(
+            region=region, projection="x1c", x=1.2, y=2.4, position="MC", text="text"
+        )
+
+
+@pytest.mark.mpl_image_compare
+def test_text_position_offset_with_line(region):
+    """
+    Print text at centre middle and  four corners (top left/right, bottom
+    left/right) of map, offset by 0.5 cm, with a line drawn from the original
+    to the shifted point.
+    """
+    fig = Figure()
+    fig.text(region=region, projection="x1c", frame="a", position="CM", text="C M")
+    for position in ("TL", "TR", "BL", "BR"):
```

Comment:
Same as above:

1. Why do you put a space between C and M?
2. It's better to also test ML, MR, TC and BC.